utils/shared.js

```diff
@@ -1184,50 +1184,59 @@ const handleCancerOccurrences = async (incomingCancerOccurrenceArray, requiredOc
             }
         }
 
-        const isCancerSiteSpecified = validateCancerOccurrence(occurrence[fieldMapping.primaryCancerSite]);
-        if (!isCancerSiteSpecified) {
-            return { error: true, message: 'No primary cancer site identified.', data: [] };
+        const cancerSiteValidationObj = validateCancerOccurrence(occurrence[fieldMapping.primaryCancerSiteObject]);
+        if (cancerSiteValidationObj.error === true) {
+            return cancerSiteValidationObj;
         }
     }
 
     // Query existing occurrences for the participant
     const { getParticipantCancerOccurrences } = require('./firestore');
     const existingCancerOccurrences = await getParticipantCancerOccurrences(participantToken);
-    
+
     // Make sure the occurrence is not a duplicate - check timestamp and yes values in primary cancer sites.
     const duplicateCheckResponse = checkForDuplicateCancerOccurrences(incomingCancerOccurrenceArray, existingCancerOccurrences);
     if (duplicateCheckResponse.error) {
         return { error: true, message: duplicateCheckResponse.message, data: [] };
     }
 
     const finalizedCancerOccurrenceArray = [];
-    // Get the highest occurrence number. Each new occurrence gets a sequential occurrenceNumber (specific to the participant).
-    let highestOccurrenceNumber = getHighestOccurrenceNumber(existingCancerOccurrences);
+    let mostRecentOccurrenceNum = existingCancerOccurrences.length;
     for (const newOccurrence of incomingCancerOccurrenceArray) {
         // Finalize the occurrence data and add to the finalized array.
-        const occurrenceData = finalizeCancerOccurrenceData(newOccurrence, participantToken, participantConnectId, highestOccurrenceNumber);
+        const occurrenceData = finalizeCancerOccurrenceData(newOccurrence, participantToken, participantConnectId, mostRecentOccurrenceNum);
         finalizedCancerOccurrenceArray.push(occurrenceData);
-        highestOccurrenceNumber++;
+        mostRecentOccurrenceNum++;
     }
 
     return { error: false, message: 'Success!', data: finalizedCancerOccurrenceArray };
 }
 
 /**
- * Validate that at least one cancer site in the Occurrence object (740819233) has a value of 'yes'.
+ * Validate additional cancer site requirements in the Occurrence object (740819233).
+ * Important: fieldMapping.primaryCancerSite value has already been validated in the initial validation step.
```

Comment:
In the jsdocs, does `fieldMapping.primaryCancerSite` need to be updated to `fieldMapping.primaryCancerSiteObj` since it was renamed in the [fieldToConceptIdMapping.js ](https://github.com/episphere/connectFaas/pull/509/files#diff-0d8ec4ec095b077565be32aa751d82e00f80837c8c53aeb2d26e37ffe2504481R209)?

Reply:
Done, thanks!